packages/invoices/src/OutboundShipment/DetailView/reducer.ts

```diff
@@ -170,33 +161,49 @@ export const reducer = (
 
           Object.keys(draft).forEach(key => {
             // TODO: Sometimes we want to keep the user entered values?
-            if (key === 'lines') return;
+            if (key === 'items') return;
             draft[key] = data[key];
           });
 
-          draft.lines = data.lines?.map(serverLine => {
-            const draftLine = draft.lines.find(
-              line => line.id === serverLine.id
-            );
-
-            if (draftLine) {
-              return mergeLines(serverLine, draftLine);
-            }
-
-            return createLine(serverLine, draft, dispatch);
-          });
-
-          draft.update = (key, value) => {
-            dispatch?.(OutboundAction.updateInvoice(key, value));
-          };
-
           draft.upsertLine = line =>
             dispatch?.(OutboundAction.upsertLine(line));
 
           draft.deleteLine = line =>
             dispatch?.(OutboundAction.deleteLine(line));
 
-          draft.items = createSummaryItems(draft.lines);
+          draft.items = data.lines?.reduce((itemsArray, serverLine) => {
+            const outboundShipmentRow = createLine(serverLine, draft);
+
+            const { existingRow, existingSummaryItem } = getExistingLine(
+              itemsArray,
+              outboundShipmentRow
+            );
+
+            const summaryItem =
+              existingSummaryItem ??
+              createSummaryItem(serverLine.itemId, [outboundShipmentRow]);
+
+            if (existingRow) {
+              delete summaryItem.batches[existingRow.id];
+              const newLine = mergeLines(serverLine, existingRow);
+              summaryItem.batches[newLine.id] = newLine;
+            } else {
+              summaryItem.batches[outboundShipmentRow.id] = outboundShipmentRow;
+            }
+
+            const { unitQuantity, numberOfPacks } =
+              recalculateSummary(summaryItem);
+
+            if (!existingSummaryItem) {
+              itemsArray.push({
+                ...summaryItem,
+                unitQuantity,
+                numberOfPacks,
+              });
+            }
+
+            return itemsArray;
+          }, draft.items);
```

Comment:
this is very similar, however we need to attach the merged lines from the server into a summary item, also